shared/src/main/scala/fr/hmil/roshttp/tools/io/IO.scala

```diff
@@ -65,13 +65,17 @@ private[roshttp] object IO {
 
   /** Reads the entire content of an input stream as a byte array. */
   def readInputStreamToByteArray(stream: InputStream): Array[Byte] = {
+    require(stream != null, "stream must not be null")
```

Comment:
Well I'd argue that InputStream is a Java API and can be null at any time since most of the time you'll be getting one from Java, so it's better to check it. See case in point. But it's your project.